internal/factsengine/gatherers/sapcontrol.go

```diff
@@ -89,24 +91,58 @@ type SapControlInstance struct {
 type SapControlGatherer struct {
 	webService sapcontrolapi.WebServiceConnector
 	fs         afero.Fs
+	cache      *factscache.FactsCache
 }
 
 func NewDefaultSapControlGatherer() *SapControlGatherer {
 	webService := sapcontrolapi.WebServiceUnix{}
 	fs := afero.NewOsFs()
-	return NewSapControlGatherer(webService, fs)
+	return NewSapControlGatherer(webService, fs, nil)
 }
 
-func NewSapControlGatherer(webService sapcontrolapi.WebServiceConnector, fs afero.Fs) *SapControlGatherer {
+func NewSapControlGatherer(
+	webService sapcontrolapi.WebServiceConnector,
+	fs afero.Fs,
+	cache *factscache.FactsCache) *SapControlGatherer {
+
 	return &SapControlGatherer{
 		webService: webService,
 		fs:         fs,
+		cache:      cache,
+	}
+}
+
+func (s *SapControlGatherer) SetCache(cache *factscache.FactsCache) {
+	s.cache = cache
+}
+
+func memoizeSapcontrol(args ...interface{}) (interface{}, error) {
+	ctx, ok := args[0].(context.Context)
+	if !ok {
+		return nil, ImplementationError.Wrap("error using memoizeSapcontrol. Context must be 1st argument")
+	}
+
+	webService, ok := args[1].(sapcontrolapi.WebServiceConnector)
+	if !ok {
+		return nil, ImplementationError.Wrap("error using memoizeSapcontrol. WebServiceConnecter must be 2nd argument")
+	}
+
+	instanceNumber, ok := args[2].(string)
+	if !ok {
+		return nil, ImplementationError.Wrap("error using memoizeSapcontrol. string must be 3rd argument")
```

Comment:
Capitalize after `.` (also in line 137)

Reply:
This is intentional. `string` type is in small case. The others are in Capital case, as they are described this way in the code